R/shared-methods.R

```diff
@@ -100,14 +100,38 @@ DataClass <- R6::R6Class(
 
     #' @description General function for downloading raw data.
     download = function() {
-      self$region$raw <- suppressWarnings(csv_reader(self$data_url))
+      if (self$verbose) {
+        message(paste(
+          "- Downloading level",
+          self$level,
+          "data for",
+          self$country
+        ))
+        self$region$raw <- suppressWarnings(
+          csv_reader(self$data_url)
+        )
+      } else {
+        self$region$raw <- suppressMessages(
+          suppressWarnings(
+            csv_reader(self$data_url)
+          )
+        )
+      }
     },
 
     #' Shared regional dataset processing
     #'
     #' @description General function to processes regional data.
     #' Dynamically works for level 1 and level 2 regions.
     process = function() {
+      if (self$verbose) {
+        message(paste(
+          "- Processing level",
```

Comment:
again why -?

Reply:
we probably don't need the country name in here as its in the early progress message? What do you think?

Reply:
Agreed